opentracing-api/src/main/java/io/opentracing/ScopeManager.java

```diff
@@ -58,12 +59,43 @@
      *     }
      * </code></pre>
      *
+     * <p>
+     * Note: You may only activate spans when you own its life cycle.
+     * That means you must make sure that no other thread calls {@link Span#finish()}
+     * while the scope is still active.
+     * If you can't guarantee that, use {@link #activate(SpanContext)} instead.
+     *
      * @param span the {@link Span} that should become the {@link #activeSpan()}
      * @return a {@link Scope} instance to control the end of the active period for the {@link Span}. It is a
      * programming error to neglect to call {@link Scope#close()} on the returned instance.
      */
     Scope activate(Span span);
```

Comment:
would be nice to deprecate this, but I expect this to not be a popular suggestion

Reply:
I think we should have a separate discussion about this

Reply:
I'd be (initially) against it ;) And yeah, this can definitely happen on a different discussion.